#! /bin/sh -e

# DP: Compile vbox using -fno-strict-aliasing.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- vbox/Makefile.in	2005-01-08 17:23:43.000000000 +0100
+++ vbox/Makefile.in~	2005-01-08 17:22:59.000000000 +0100
@@ -53,7 +53,7 @@
 #----------------------------------------------------------------------------#
 
 CC		= @CC@
-CFLAGS		= @CFLAGS@ @LINK_NCURSES_INCL@ @LINK_TCL_INCL@
+CFLAGS		= @CFLAGS@ @LINK_NCURSES_INCL@ @LINK_TCL_INCL@ -Wall -fno-strict-aliasing
 LDFLAGS		= @LDFLAGS@
 
 #----------------------------------------------------------------------------#
